macros/models/_sector/tokens/transfers_base.sql

```diff
@@ -0,0 +1,110 @@
+
+WITH transfers AS (
+    SELECT block_time
+    , block_number
+    , tx_hash
+    , cast(NULL as bigint) AS evt_index
+    , trace_address
+    {% if native_contract_address%}
+    , {{native_contract_address}} AS contract_address
+    {% else %}
+    , CAST(NULL AS varbinary) AS contract_address
+    {% endif %}
+    , 'native' AS token_standard
+    , "from"
+    , to
+    , value AS amount_raw
+    FROM {{ traces }}
+    WHERE success
+    AND (call_type NOT IN ('delegatecall', 'callcode', 'staticcall') OR call_type IS null)
+    AND value > UINT256 '0'
+    {% if is_incremental() %}
+    AND {{incremental_predicate('block_time')}}
+    {% endif %}
+
+    UNION ALL
+
+    SELECT t.evt_block_time AS block_time
+    , t.evt_block_number AS block_number
+    , t.evt_tx_hash AS tx_hash
+    , t.evt_index
+    , CAST(NULL AS ARRAY<BIGINT>) AS trace_address
+    , t.contract_address
+    , '{{token_standard_20}}' AS token_standard
+    , t."from"
+    , t.to
+    , t.value AS amount_raw
+    FROM {{ erc20_transfers }} t
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('evt_block_time')}}
+    {% endif %}
+
+    {% if wrapped_token_deposit and wrapped_token_withdrawal %}
+    UNION ALL
+
+    SELECT t.evt_block_time AS block_time
+    , t.evt_block_number AS block_number
+    , t.evt_tx_hash AS tx_hash
+    , t.evt_index
+    , CAST(NULL AS ARRAY<BIGINT>) AS trace_address
+    , t.contract_address
+    -- technically this is not a standard 20 token, but we use it for consistency
+    , '{{token_standard_20}}' AS token_standard
+    , 0x0000000000000000000000000000000000000000 as "from"  -- TODO: change to variable
+    , t.dst as "to"
+    , t.wad AS amount_raw -- is this safe cross chain?
+    FROM {{ wrapped_token_deposit }} t
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('evt_block_time')}}
+    {% endif %}
+
+    UNION ALL
+
+    SELECT t.evt_block_time AS block_time
+    , t.evt_block_number AS block_number
+    , t.evt_tx_hash AS tx_hash
+    , t.evt_index
+    , CAST(NULL AS ARRAY<BIGINT>) AS trace_address
+    , t.contract_address
+    -- technically this is not a standard 20 token, but we use it for consistency
+    , '{{token_standard_20}}' AS token_standard
+    , t.src as "from"
+    , 0x0000000000000000000000000000000000000000 as "to"  -- TODO: change to variable
+    , t.wad AS amount_raw -- is this safe cross chain?
+    FROM {{ wrapped_token_withdrawal }} t
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('evt_block_time')}}
+    {% endif %}
+{% endif %}
+    )
+
+SELECT '{{blockchain}}' as blockchain
+, cast(date_trunc('day', t.block_time) as date) as block_date
+, t.block_time
+, t.block_number
+, t.tx_hash
+-- method_id => first 4bytes of data
+-- We have to create this because evt_index and trace_address can be null
+, {{dbt_utils.generate_surrogate_key(['t.block_number', 'tx.index', 't.evt_index', "array_join(trace_address, ',')"])}} as unique_key
```

Comment:
do we prefer this approach of adding surrogate key, rather than `coalesce()` each unique key to a default value?

this could be a decision that has an impact across spellbook. of course, we don't have to standardize it all right away, but will be nice to be consistent moving forward at minimum (and on larger sector spells)

Reply:
TBH I don't like having `unique_key` in most cases but here I think it works because not sure what we should coalesce to. I think it can be confusing for users to have a value in `evt_index/trace_address` that doesn't lead anywhere. 

I think if we do have a unique key column we should always try to use `generate_surrogate_key` because 1) it hashes the output so that it takes less space and 2) it works the same everywhere.

Reply:
i'm okay with the rationale behind the first part. i will look to consider the approach in new `dex.trades` design, if applicable.

agree fully with the second part.

Reply:
I still believe we should adapt the incremental logic to use `IS NOT DISTINCT FROM`  as equality operation so that the merge works as we would want with respect to `null` values.